File: vepserver/src/main/scala/vep/model/session/Session.scala
package vep.model.session

import org.joda.time.DateTime

case class Session(id: Int, theater: Int, canonical: String, date: DateTime, name: String, reservationEndDate: DateTime)

case class SessionPrice(id: Int, session: Int, name: String, price: Int, cases: Option[String])

case class SessionShow(session: Int, show: Int, num: Int)

case class SessionDetailParsed(id: Int, theater: String, canonical: String, date: DateTime, name: String, reservationEndDate: DateTime) {
  def toSessionDetail(prices: Seq[SessionPriceDetail], shows: Seq[String]) =
    SessionDetail(theater, canonical, date, name, reservationEndDate, prices, shows)
}

case class SessionDetail(theater: String, canonical: String, date: DateTime, name: String, reservationEndDate: DateTime,
                         prices: Seq[SessionPriceDetail], shows: Seq[String])

case class SessionPriceDetail(id: Int, name: String, price: Int, condition: Option[String])

File: vepweb/lib/model/session/Session.dart
part of vep.model.session;

class Session extends ModelToString {
  String theater;
  String canonical;
  DateTime date;
  String name;
  DateTime reservationEndDate;
  List<SessionPrice> prices;
  List<String> shows;

  SessionForm toSessionForm([SessionForm sessionForm]) {
    if (sessionForm == null) {
      sessionForm = new SessionForm();
    }
    copyByAccessors(this, sessionForm);
    sessionForm.internalShows = sessionForm.shows == null ? null : sessionForm.shows.map((_){
      var sessionFormShow = new SessionFormShow();
      sessionFormShow.show = _;
      return sessionFormShow;
    }).toList();
    return sessionForm;
  }

  SessionFormUpdate toSessionFormUpdate() {
    return toSessionForm(new SessionFormUpdate());
  }
}

class SessionPrice extends ModelToString {
  String name;
  int price;
  String condition;
}

class SessionForm extends ModelToString {
  @jsonx.jsonIgnore
  String theater;
  @jsonx.jsonIgnore
  String canonical;

  DateTime date;
  String name;
  DateTime reservationEndDate;
  List<SessionPrice> prices;
  List<String> shows;

  @jsonx.jsonIgnore
  List<SessionFormShow> internalShows;

  String get strDate => date != null ? new DateFormat('dd/MM/yyyy HH:mm').format(date) : '';
  String get strReservationEndDate => reservationEndDate != null ? new DateFormat('dd/MM/yyyy HH:mm').format(reservationEndDate) : '';
}

class SessionFormUpdate extends SessionForm {
  String reason;
}

class SessionFormShow extends ModelToString {
  String show;
}